packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -669,19 +722,19 @@ rocksdb::Status ArbCore::reorgCheckpoints(
         }
     }
 
+    ReadWriteTransaction tx(data_storage);
     uint256_t next_sideload_block_number = 0;
-    if (output.last_sideload) {
+    if (output.last_sideload.has_value()) {
         next_sideload_block_number = *output.last_sideload + 1;
     }
 
-    ReadWriteTransaction tx(data_storage);
     auto status = deleteSideloadsStartingAt(tx, next_sideload_block_number);
     if (!status.ok()) {
         return status;
     }
 
     // Delete logs individually to handle reference counts
-    auto optional_status = deleteLogsStartingAt(tx, output.log_count);
+    auto optional_status = deleteLogsStartingAt(tx, output.log_count + 1);
```

Comment:
I think the previous version was correct, as deleteLogsStartingAt takes an index as its argument.